### PR TITLE
renovate: Replace update-hubble-version.sh with Renovate Bot

### PR DESCRIPTION
**Pull request by gandro**
TL;DR: With this, Cilium will automatically pull in the newest Hubble CLI version in all stable branches (and master) whenever there is a new Hubble CLI GitHub release and/or Docker image.

---

This commit removes the update-hubble-version.sh script and replaces it with two Renovate configuration entries.

The first entry, `kubernetes`, is using Renovate's standard `kubernetes` and `docker` managers to update the Hubble CLI Docker image in `examples/hubble/hubble-cli.yaml` whenever there is a newer Hubble CLI docker image. To make this take effect, we have to set the `ignorePaths` to an empty list, because the `config:base` preset ignores any paths that contain an `/examples/` directory: https://docs.renovatebot.com/presets-default/#ignoremodulesandtests 

Setting the `ignorePaths` to the empty list is safe, because our configuration relies on `includePaths` instead.  Additionally, because Renovate only updates the Docker digest if it is prsent, this commit also pins the Hubble CLI docker image digest so that it is updated by Renovate in subsequent PRs.

The second new entry in the Renovate config is a custom `regexManager`. It defines two custom regexes to bump the version and digest of the Hubble CLI GitHub release. The syntax is inspired by Renovate's own Regex Manager Presets defined here: https://docs.renovatebot.com/presets-regexManagers/

The digest update is the most interesting part of this PR. Renovate's `github-releases` manager has a sophisticated mechanism to update digests of artifacts, without any direct reference to artifacts name or architecture. The way this works is that Renovate will fetch all digest files in the old release, find the digest file which contains the `currentDigest` sha, and then replace that digest sha with the one found in the matching artifact file of new release. For this to work, we need to store what version the digest is coming from, thus the custom regex also contains a `digestVersion` field. For more information, see the [mapDigestAssetToRelease](https://github.com/renovatebot/renovate/blob/461cbc75518946fe6c7612122f9ccb2e8202e583/lib/modules/datasource/github-releases/index.ts#L127-L128) function in Renovate's source code.

Lastly, this commit removes the `update-hubble-version.sh` script to avoid diverging update mechanisms. Because we no longer need to generate a `hubble-version.sh` file, that file is also now inlined directly by `download-hubble.sh`.

I have tested this commit on my fork of cilium/cilium. Renovate (in self-hosting mode) created the following two PRs:

  - https://github.com/gandro/cilium/pull/85
  - https://github.com/gandro/cilium/pull/86

### CODE REVIEW

**Review by rolinh — APPROVED** [review]
This is pretty awesome!

**Review by kaworu — APPROVED** [review]
Thanks @gandro nice automation and cleanup :robot:

**Review by aanm — APPROVED** [review]

**Review by qmonnet — APPROVED** [review]
Looks cool!

**Comment by qmonnet** [comment]
Travis hit https://github.com/cilium/cilium/issues/23509
